server/models/user.js

```diff
@@ -98,12 +98,40 @@ userSchema.methods.validPassword = function(password) {
   return bcrypt.compareSync(password, this.password);
 };
 
+userSchema.methods.getMessagesSeparated = function(next) {
+  this.model('User').findById(this.id).deepPopulate([
+    'messages',
+    'messages.sender'
+  ]).exec((err, user) => {
+    var oldMessages = new Set();
+    var newMessages = new Set();
+    for(var i = 0; i < user.messages.length; i++) {
```

Comment:
nope, ovo je antipattern, nikada u JS-u neces pisati ovakve stvari. Imas jednostavnije strukture i metoda koje ti ovo omogućuju. 
npr:
```
const messages = user.messages.filter((message) => message.sender.id === this.id);
const newMessages = messages.filter((message) => !message.isSeen)
const oldMessages = messages.filter((message) => message.isSeen)
```
mislim da je to ista logika kao i ova tvoja, a malo je citljivija jer vidis da prvo uzimas samo poruke kojima je sender jedak trenutnom useru a zatim ih grupiras na new i old ovisno jesu li isSeen.
drugi nacin je da ides 
```
const newMessages = [];
const oldMessages = [];
messages.forEach((message) => {
  if (message.isSeen) {
    oldMessages.push(message);
  } else {
    newMessages.push(message);
  }
});
```